src/mon/PGMonitor.h

```diff
@@ -32,16 +32,15 @@ using namespace std;
 #include "msg/Messenger.h"
 #include "mon/MonitorDBStore.h"
 
-class MStatfs;
-class MMonCommand;
-class MGetPoolStats;
-
-class TextTable;
 class MPGStats;
+class PGStatService;
 
 class PGMonitor : public PaxosService {
-public:
   PGMap pg_map;
+  std::unique_ptr<PGStatService> pgservice;
+
+  bool do_delete = false;   ///< propose deleting pgmap data
+  bool did_delete = false;  ///< we already deleted pgmap data
```

Comment:
It's a one-time event so I'm not sure getting Monitor involved will help.  I'm also trying to keep the hackery confined to PGMonitor as much as possible.

I like the idea of making it crash out, though.. maybe pgmon() could return null once the switch is flipped?